client/signup/steps/user/index.jsx

```diff
@@ -149,6 +150,7 @@ export class UserStep extends Component {
 				submitForm={ this.submitForm }
 				submitButtonText={ this.submitButtonText() }
 				suggestedUsername={ this.props.suggestedUsername }
+				isSocialSignupEnabled={ isEnabled( 'signup/social' ) && this.props.flowName === 'social' }
```

Comment:
It's not necessary to duplicate the whole step component. Different steps can share the same component, for example: https://github.com/Automattic/wp-calypso/blob/16d8d9c5388c507eb8dd7c6d8988793dfc4609fa/client/signup/config/steps.js#L92